packages/nimbus/src/components/collapsible-motion/components/collapsible-motion-trigger.tsx

```diff
@@ -0,0 +1,48 @@
+import { forwardRef } from "react";
+import { mergeProps } from "react-aria";
+import { useCollapsibleMotionContext } from "./collapsible-motion-context";
+import { CollapsibleMotionTriggerSlot } from "../collapsible-motion.slots";
+import type { CollapsibleMotionTriggerProps } from "../collapsible-motion.types";
+import { Button as RaButton } from "react-aria-components";
+import { chakra } from "@chakra-ui/react/styled-system";
+import { extractStyleProps } from "@/utils/extractStyleProps";
+
+/**
+ * CollapsibleMotion.Trigger - The trigger button that toggles the collapsible content
+ *
+ * This component renders a button with proper ARIA attributes for accessibility.
+ * It automatically handles the toggle functionality when clicked.
+ */
+export const CollapsibleMotionTrigger = forwardRef<
+  HTMLButtonElement,
+  CollapsibleMotionTriggerProps
+>(({ children, onClick, asChild, ...props }, forwardedRef) => {
+  const { buttonProps, isDisabled } = useCollapsibleMotionContext();
+
+  // Use mergeProps to properly compose event handlers like React Aria does
+  const componentProps = mergeProps(props, buttonProps, {
+    disabled: isDisabled,
+    onClick, // Consumer's custom onClick handler
```

Comment:
why are we supporting `onClick` here?  Isn't the idea that the child will be a flavor of the nimbus button (eg an `IconButton` instead of a `Button`) that uses `onPress`?